src/components/search/tourSteps/CompareTourSteps.tsx

```diff
@@ -0,0 +1,122 @@
+import { PaginationObject } from 'app/Types';
+
+/* Import Sources */
+import SearchTourStepsText from 'sources/tourText/search.json';
+
+
+/* Props Type */
+type Props = {
+    pagination: PaginationObject
+};
+
+
+/**
+ * Component that renders the tour steps for the homepage
+ * @param pagination The pagination object used to index the digital specimen results
+ * @returns JSX Component
+ */
+const CompareTourSteps = (props: Props) => {
+    const { pagination } = props;
+
+    /* Hooks */
+    const dispatch = useAppDispatch();
+    const searchFilters = useSearchFilters();
+
+    /* Base variables */
+    const tourTopic = useAppSelector(getTourTopic);
+    const compareDigitalSpecimen = useAppSelector(getCompareDigitalSpecimen);
+    const compareSteps = SearchTourStepsText.compare;
+    const { options } = StepsConfig();
+    let steps: {
+        intro: string,
+        element?: string
+    }[] = [];
+
+    /* Construct Intro.js steps for Homepage */
+    compareSteps.forEach((step, index) => {
+        steps.push({
+            intro: step,
+            element: `.tourCompare${index + 1}`
+        });
+    });
+
+    return (
+        <Steps enabled={tourTopic === 'compare'}
+            steps={steps}
+            initialStep={0}
+            onBeforeChange={(nextIndex) => {
+                return new Promise((resolve) => {
+                    if ([0, 1].includes(nextIndex)) {
+                        dispatch(setCompareDigitalSpecimen(undefined));
+
+                        resolve();
+                    } else if (nextIndex === 2) {
+                        /* On step 3: set search query to 'bellis perennis' and enable comparison mode */
+                        searchFilters.SetSearchFilters({
+                            q: 'bellis perennis'
+                        });
+
+                        dispatch(setCompareDigitalSpecimen([]));
+
+                        resolve();
+                    } else if (nextIndex === 3) {
+                        /* On step 4: simulate selecting specimen for comparison */
+                        if (!compareDigitalSpecimen?.length) {
+                            dispatch(setCompareDigitalSpecimen([pagination.records[0] as DigitalSpecimen]));
+
+                            resolve();
+                        } else if (compareDigitalSpecimen.length) {
+                            const localCompareDigitalSpecimen = [...compareDigitalSpecimen];
+                            localCompareDigitalSpecimen.pop();
+
+                            dispatch(setCompareDigitalSpecimen([...localCompareDigitalSpecimen]));
+
+                            resolve();
+                        } else {
+                            resolve();
+                        }
+                    } else if ([4, 5].includes(nextIndex)) {
+                        /* On step 5: simulate selecting of second specimen for comparison */
+                        const compareDigitalSpecimen: DigitalSpecimen[] = [];
```

Comment:
I think this shadows a variable from outside scope

Reply:
Yes but that does not matter in this context, as it is in another scope than the compareDigitalSpecimen variable at the top